### PR TITLE
Added rate limiter to comms stress test

### PR DESCRIPTION
**Pull request by hansieodendaal**
## Description
Adding a rate limiter to the comms stress test.

## Motivation and Context
Need to stress test the yamux `MAX_BUFFER_SIZE` and `RECEIVE_WINDOW` while stressing the comms.

## How Has This Been Tested?
Local test between 2 test executables.

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [X] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [X] I'm merging against the `development` branch.
* [X] I ran `cargo-fmt --all` before pushing.
* [X] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: comms/examples/stress_test.rs [hunk 1]
```diff
@@ -56,6 +56,13 @@ async fn run() -> Result<(), Error> {
 
     let is_tcp = remove_arg(&mut args, "--tcp").is_some();
 
+    let mut rate_limit = 100_000;
+    println!("Default rate limit is: {}/s", rate_limit);
+    if let Some(pos) = remove_arg(&mut args, "--rate-limit") {
+        rate_limit = args.remove(pos).parse().expect("Unable to parse rate limit");
+        println!("Rate limit reset to:   {}/s", rate_limit);
+    }
+
```
Comment (sdbondi) [inline comms/examples/stress_test.rs]: Why not just output what the rate limit is set to?

```suggestion
    let mut rate_limit = 100_000;
    if let Some(pos) = remove_arg(&mut args, "--rate-limit") {
        rate_limit = args.remove(pos).parse().expect("Unable to parse rate limit");
    }
        println!("Rate limit set to:   {}/s", rate_limit);
```

File: comms/examples/stress/service.rs [hunk 2]
```diff
@@ -44,10 +44,17 @@ use tari_comms::{
 };
 use tari_crypto::tari_utilities::hex::Hex;
 use tokio::{task, task::JoinHandle};
+use tari_comms::rate_limit::RateLimit;
+use tokio::time::Duration;
+
+/// The minimum amount of inbound messages to accept within the `RATE_LIMIT_RESTOCK_INTERVAL` window
+const RATE_LIMIT_MIN_CAPACITY: usize = 1;
```
Comment (sdbondi) [inline comms/examples/stress/service.rs:51]: This is not the minimum capacity - it's just the capacity
```suggestion
const RATE_LIMIT_CAPACITY: usize = 1;
```

**Review by sdbondi — COMMENTED** [review]
LGTM -  one nit

File: comms/examples/stress/service.rs [hunk 3]
```diff
@@ -44,10 +44,17 @@ use tari_comms::{
 };
 use tari_crypto::tari_utilities::hex::Hex;
 use tokio::{task, task::JoinHandle};
+use tari_comms::rate_limit::RateLimit;
+use tokio::time::Duration;
```
Comment (sdbondi) [inline comms/examples/stress/service.rs]:
```suggestion
use std::time::Duration;
```

**Review by sdbondi — APPROVED** [review]
LGTM

**Review by stringhandler — APPROVED** [review]
Didn't test it locally, but looks good from code review

**Pull request merged**